test/controllers/todos_controller_test.rb

```diff
@@ -989,6 +989,17 @@ def test_url_with_slash_in_query_string_are_parsed_correctly
     assert_select "a[href='#{url}']"
   end
 
+  def test_url_with_double_hyphen_is_parsed_correctly
```

Comment:
In general we have way too many tests at the wrong level or in the wrong spot. We should aim to true that up as we refactor. For the moment it's a moot point here, though.